site/content/entry/citra-progress-report-2017-september.md

```diff
@@ -0,0 +1,360 @@
+are more likely to only need data already in the cache. Textures that have been
+Morton coded are usually referred to as swizzled or twiddled textures.
+
+{{< figure src="/images/entry/citra-progress-report-2017-september/morton-koopa.png#floatright" 
+    title="Not this Morton!" >}}
+
+In the function that Morton is implemented, there was a lookup table on Morton
+codes in the comments, and [huwpascoe](https://github.com/huwpascoe) thought it'd
+be best if we just use the lookup table directly. It worked just as well as before,
+but required less than a third of the math. Because this function is called so
+often during emulation (a rough estimate from them is about "millions of times a
+second"), this change although small, made very big changes in CPU performance.
+
+## [Add draw for immediate and batch modes](https://github.com/citra-emu/citra/pull/2921) by [jroweboy](https://github.com/jroweboy)
+
+The 3DS' GPU has two main modes for drawing to the screen, immediate and batch
+mode. In the former, the GPU takes and immediately draws every vertex as it is
+handed to it. In the latter, the GPU accepts vertices given to it, but doesn't
+actually bother drawing them until absolutely necessary, saving a bit of time
+from not having to go through the drawing procedure for every individual vertex.
+Although most games don't use immediate mode at all due to it being extremely
+slow, a handful do use it for a handful of visual effects, like New Super Mario
+Bros. 2.
+
+About a year ago when the GPU code on Citra was rewritten, a handful of calls to
+the drawing routine were removed, as it was believed they were unnecessary. Turns
+out, one of the calls was actually needed for some effects in games, as it handled
+immediate mode drawing. This wasn't noticed for a very long time, as most games
+appeared to carry on with no side-effects at all from the rewrite, but was eventually
+found after some research courtesy of [ds84182](https://github.com/ds84182).
+
+## [Interpolate audio samples on a frame-by-frame basis](https://github.com/citra-emu/citra/pull/2858) by [MerryMage](https://github.com/MerryMage)
+
+When a 3DS game needs some sort of audio processing, they can access the 3DS' DSP,
+or __D__igital __S__ound __P__rocessor. It's another processor, alongside the ARM9
+and ARM11, that is given a firmware to run, which in turn is given a bunch of audio
+samples and parameters by the game. The DSP then plays back the buffer in chunks
+of about 5 milliseconds. Each one of these chunks is called an audio frame.
+
+As of today, we don't know how the DSP exactly works, and we don't know how any
+of the firmwares exactly work. (Did I forget to mention earlier there's multiple
+versions of the firmware?) But we do know how to use it, and from there we can
+reimplement its behaviour directly in Citra. Which is exactly what [MerryMage](https://github.com/MerryMage)
+did, which in turn brought audio support for the first time in Citra.
```

Comment:
This sentence seems to imply that audio support was brought in September, IMO. Something like "This is the approach MerryMage took to implement initial audio support back in June of 2016" (or something)

Reply:
(could also link the blog post for this!)

Reply:
There's a blog post?